File: SEPTEMBER-2025/11-09-2025/p2785.py
class Solution:
    def sortVowels(self, s: str) -> str:
        stack = []
        vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}

        for x in s:
            if x in vowels:
                stack.append(x)

        stack.sort(reverse=True)
        res = ''

        for x in s:
            if x in vowels:
                res += stack.pop()
            else:
                res += x

        return res

File: SEPTEMBER-2025/11-09-2025/p2785.cpp
class Solution {
public:
    string sortVowels(string s) {
        unordered_set<char>vowels={'E','I','O','U','A','a','e','i','o','u'};
        string temp="";
        int n=s.length();
        for(int i=0;i<n;i++){
            if(vowels.find(s[i])!=vowels.end()){
                temp.push_back(s[i]);
            }
        }
        sort(temp.begin(),temp.end());
        int j=0;
        for(int i=0;i<n;i++){
            if(vowels.find(s[i])!=vowels.end()){
                s[i]=temp[j++];
            }
        }
        return s;
    }
};